src/main/java/core/basesyntax/Rectangle.java

```diff
@@ -22,4 +22,33 @@ public Integer getLength() {
     public String getColor() {
         return color;
     }
+
+    @Override
+    public boolean equals(Object rectangle) {
+        if (rectangle == this) {
+            return true;
+        }
+        if (rectangle == null) {
+            return false;
+        }
+        Rectangle castedRectangle = (Rectangle) rectangle;
+        if (!castedRectangle.getClass().toString().equals(Rectangle.class.toString())) {
+            return false;
+        }
+        return ((this.color == null && castedRectangle.color == null)
+                || this.color.equals(castedRectangle.color))
+                && ((this.width == null && castedRectangle.width == null)
+                || this.width.equals(castedRectangle.width))
+                && ((this.length == null && castedRectangle.length == null)
+                || this.length.equals(castedRectangle.length));
```

Comment:
1) let's not use `this`

2) if both `color` and `castedRectangle.color` is null, the following statement will be `true` - `color == castedRectangle.color`

3) what if `color` is null and `castedRectangle.color` is not null?